teos/chain_monitor.py

```diff
@@ -59,38 +78,29 @@ def __init__(self, watcher_queue, responder_queue, block_processor, bitcoind_fee
             )
         )
 
-        self.watcher_queue = watcher_queue
-        self.responder_queue = responder_queue
+        self.receiving_queues = receiving_queues
 
         self.polling_delta = 60
         self.max_block_window_size = 10
         self.block_processor = block_processor
+        self.queue = Queue()
+        self.status = ChainMonitorStatus.IDLE
 
-    def notify_subscribers(self, block_hash):
-        """
-        Notifies the subscribers (``Watcher`` and ``Responder``) about a new block. It does so by putting the hash in
-        the corresponding queue(s).
-
-        Args:
-            block_hash (:obj:`str`): the new block hash to be sent to the subscribers.
-        """
-
-        self.watcher_queue.put(block_hash)
-        self.responder_queue.put(block_hash)
-
-    def update_state(self, block_hash):
+    def enqueue(self, block_hash):
```

Comment:
Tbh I don't think it hurts to have it (it was like this also before).
Checking the queue length after enqueuing might be unreliable if the thread that empties the queue is active, as it might remove an element before the check.

Reply:
OK, fair enough